allennlp/modules/token_embedders/bidirectional_token_embedder.py

```diff
@@ -0,0 +1,112 @@
+from typing import Dict
+
+import torch
+
+from allennlp.common.checks import ConfigurationError
+from allennlp.modules import ScalarMix
+from allennlp.modules.text_field_embedders import TextFieldEmbedder
+from allennlp.modules.token_embedders.token_embedder import TokenEmbedder
+from allennlp.modules.seq2seq_encoders import BidirectionalLanguageModelTransformer
+from allennlp.nn.util import device_mapping, get_text_field_mask, remove_sentence_boundaries
+
+
+@TokenEmbedder.register('bidirectional_token_embedder')
+class BidirectionalTokenEmbedder(TokenEmbedder):
```

Comment:
is "BidirectionalTokenEmbedder" the best name for this? I would have expected the name to mention "language model" and possibly "transformer"

Reply:
Yeah, that's fair. Changed to `BidirectionalLanguageModelTokenEmbedder`. Just unfortunate that this is getting a  bit unwieldy... Strictly speaking a transformer contextualizer is no longer required -- just one that supports `num_layers` and `return_all_layers`. I've documented this in the constructor here.